### ISSUE TITLE
Need help on how to read properties and value (and write them)

### ISSUE THREAD

**Issue by Apollon77** [fh1ch/node-bacstack#147]
Node Version: `12.x`

Node BACstack Version: `current GitHub Master branch`

- [ ] Bug Report
- [ ] Feature Request
- [x] Question

### Question
I'm currently trying to get familiar with Bacnet because I want to implement this into the open source smart home platform ioBroker.

I currently use the compliance docker image with the "official" test device from the original backsack c library.

I managed to read the object details using the example provided at https://fh1ch.github.io/node-bacstack/bacstack.html#.readPropertyMultiple . The result is

```
{"len":651,"values":[{"objectId":{"type":8,"instance":1234},"values":[{"id":75,"index":4294967295,"value":[{"value":{"type":8,"instance":1234},"type":12}]},{"id":77,"index":4294967295,"value":[{"value":"SimpleServer","type":7,"encoding":0}]},{"id":79,"index":4294967295,"value":[{"value":8,"type":9}]},{"id":112,"index":4294967295,"value":[{"value":0,"type":9}]},{"id":121,"index":4294967295,"value":[{"value":"BACnet Stack at SourceForge","type":7,"encoding":0}]},{"id":120,"index":4294967295,"value":[{"value":260,"type":2}]},{"id":70,"index":4294967295,"value":[{"value":"GNU","type":7,"encoding":0}]},{"id":44,"index":4294967295,"value":[{"value":"0.8.5","type":7,"encoding":0}]},{"id":12,"index":4294967295,"value":[{"value":"1.0","type":7,"encoding":0}]},{"id":98,"index":4294967295,"value":[{"value":1,"type":2}]},{"id":139,"index":4294967295,"value":[{"value":12,"type":2}]},{"id":97,"index":4294967295,"value":[{"value":{"value":[233,208,19,84,159],"bitsUsed":40},"type":8}]},{"id":96,"index":4294967295,"value":[{"value":{"value":[63,229,58,16,0,128,1],"bitsUsed":51},"type":8}]},{"id":76,"index":4294967295,"value":[{"value":{"type":8,"instance":1234},"type":12},{"value":{"type":0,"instance":0},"type":12},{"value":{"type":0,"instance":1},"type":12},{"value":{"type":0,"instance":2},"type":12},{"value":{"type":0,"instance":3},"type":12},{"value":{"type":1,"instance":0},"type":12},{"value":{"type":1,"instance":1},"type":12},{"value":{"type":1,"instance":2},"type":12},{"value":{"type":1,"instance":3},"type":12},{"value":{"type":2,"instance":0},"type":12},{"value":{"type":2,"instance":1},"type":12},{"value":{"type":2,"instance":2},"type":12},{"value":{"type":2,"instance":3},"type":12},{"value":{"type":3,"instance":0},"type":12},{"value":{"type":3,"instance":1},"type":12},{"value":{"type":3,"instance":2},"type":12},{"value":{"type":3,"instance":3},"type":12},{"value":{"type":3,"instance":4},"type":12},{"value":{"type":4,"instance":0},"type":12},{"value":{"type":4,"instance":1},"type":12},{"value":{"type":4,"instance":2},"type":12},{"value":{"type":4,"instance":3},"type":12},{"value":{"type":5,"instance":0},"type":12},{"value":{"type":5,"instance":1},"type":12},{"value":{"type":5,"instance":2},"type":12},{"value":{"type":5,"instance":3},"type":12},{"value":{"type":5,"instance":4},"type":12},{"value":{"type":5,"instance":5},"type":12},{"value":{"type":5,"instance":6},"type":12},{"value":{"type":5,"instance":7},"type":12},{"value":{"type":5,"instance":8},"type":12},{"value":{"type":5,"instance":9},"type":12},{"value":{"type":15,"instance":0},"type":12},{"value":{"type":15,"instance":1},"type":12},{"value":{"type":21,"instance":0},"type":12},{"value":{"type":21,"instance":1},"type":12},{"value":{"type":21,"instance":2},"type":12},{"value":{"type":21,"instance":3},"type":12},{"value":{"type":21,"instance":4},"type":12},{"value":{"type":21,"instance":5},"type":12},{"value":{"type":21,"instance":6},"type":12},{"value":{"type":28,"instance":0},"type":12},{"value":{"type":28,"instance":1},"type":12},{"value":{"type":28,"instance":2},"type":12},{"value":{"type":28,"instance":3},"type":12},{"value":{"type":13,"instance":0},"type":12},{"value":{"type":13,"instance":1},"type":12},{"value":{"type":13,"instance":2},"type":12},{"value":{"type":13,"instance":3},"type":12},{"value":{"type":14,"instance":0},"type":12},{"value":{"type":14,"instance":1},"type":12},{"value":{"type":14,"instance":2},"type":12},{"value":{"type":14,"instance":3},"type":12},{"value":{"type":19,"instance":0},"type":12},{"value":{"type":19,"instance":1},"type":12},{"value":{"type":19,"instance":2},"type":12},{"value":{"type":19,"instance":3},"type":12},{"value":{"type":20,"instance":0},"type":12},{"value":{"type":20,"instance":1},"type":12},{"value":{"type":20,"instance":2},"type":12},{"value":{"type":20,"instance":3},"type":12},{"value":{"type":20,"instance":4},"type":12},{"value":{"type":20,"instance":5},"type":12},{"value":{"type":20,"instance":6},"type":12},{"value":{"type":20,"instance":7},"type":12},{"value":{"type":10,"instance":0},"type":12},{"value":{"type":10,"instance":1},"type":12},{"value":{"type":10,"instance":2},"type":12},{"value":{"type":47,"instance":0},"type":12},{"value":{"type":47,"instance":1},"type":12},{"value":{"type":47,"instance":2},"type":12},{"value":{"type":47,"instance":3},"type":12},{"value":{"type":48,"instance":0},"type":12},{"value":{"type":48,"instance":1},"type":12},{"value":{"type":48,"instance":2},"type":12},{"value":{"type":48,"instance":3},"type":12},{"value":{"type":17,"instance":0},"type":12},{"value":{"type":17,"instance":1},"type":12},{"value":{"type":17,"instance":2},"type":12},{"value":{"type":17,"instance":3},"type":12}]},{"id":62,"index":4294967295,"value":[{"value":1476,"type":2}]},{"id":107,"index":4294967295,"value":[{"value":3,"type":9}]},{"id":11,"index":4294967295,"value":[{"value":3000,"type":2}]},{"id":73,"index":4294967295,"value":[{"value":3,"type":2}]},{"id":30,"index":4294967295,"value":[]},{"id":155,"index":4294967295,"value":[{"value":1,"type":2}]},{"id":28,"index":4294967295,"value":[{"value":"server","type":7,"encoding":0}]},{"id":57,"index":4294967295,"value":[{"value":"1901-02-01T23:30:40.260Z","type":11}]},{"id":119,"index":4294967295,"value":[{"value":0,"type":3}]},{"id":56,"index":4294967295,"value":[{"value":"2020-06-16T00:00:00.000Z","type":10}]},{"id":24,"index":4294967295,"value":[{"value":false,"type":1}]},{"id":58,"index":4294967295,"value":[{"value":"USA","type":7,"encoding":0}]},{"id":152,"index":4294967295,"value":[]}]}]}
```

So de details for all properties of obnject list (76) look like 

```
{"value":{"type":0,"instance":2},"type":12}
```
So I understand that the "Type" means that there is analog input with instance id 2 ... but how do I get the details for that?
When reading via "bacepics" then they show data like
```
{
object-identifier: (analog-input, 2)
object-name: "ANALOG INPUT 2"
object-type: analog-input
present-value: 0.000000
status-flags: {false,false,false,false}
event-state: normal
out-of-service: FALSE
units: percent
},
```

So how can I query those using the lib? Everything I tried did not gave me meaningful results ...

Same goes with writeing one f these values ...

Basically it works for "top level" objects ...like type 121 and such, I can read them, but in fact they are not writable.

But how do I do withthe other properties?

Any help or ideas or advice would be awesome.

Ingo

**Comment by Apollon77** [comment]
I think I understood it in the meantime:

e.g. to read type 1 and instance 2 (it seems these compnations are unque also when multiple devices are on one server:

```
const requestArray = [{
      objectId: {type: 1, instance: 2},
      properties: [{id: 8}]
    }];
```
